### DISCUSSION TITLE
Concatenating variables with the same name but different configurations is hard

### DISCUSSION THREAD

**Discussion by Kiskadee-dev** [biolab/orange3#6037] [Q&A]
I am currently working with many csv datasets who have the exact same headers, I could (and did later) merge them manually but at first I thought the concatenate widget would be enough. It was this way so I could easily reload a specific dataset when updating them or correcting mistakes inside the file, with only one its a bit clumsy.. anyway, right now we don't have a way to apply the configuration of the features to multiple file imports, this can easily lead to mistakes that will mess with the concatenate widget as it won't concatenate columns with different configurations.

I think the concatenate widget could be expanded so we can remap the datasets if we wish to by manually matching the columns headers.

Working with a single new column and then trying to merge it to the dataset is also clumsy as often it'll have different configurations than the original one, some time ago i had a bad time merging two columns, 'feature' and 'predicted feature', where the predicted ones were only available when the feature wasn't. I think we need at least a better error message to explain why the concatenate widget renamed the variables and why some features ended up with '?'.

Tl; DR: Concatenating variables with the same name and keeping the same name is hard

I am new to orange.

**Comment by ajdapretnar** [comment]
Concatenate has the option "Treat variables with the same name as the same variable, even if they are computed using different formulae", which solves your issue.

That said, the documentation for Concatenate is outdated and should be rewritten. Getting it fixed asap.

**Reply by Kiskadee-dev** [reply]
![image](https://user-images.githubusercontent.com/10299707/174696538-83982365-293a-41d0-9190-1c9ced711fae.png)
![image](https://user-images.githubusercontent.com/10299707/174696564-dd16fdb3-e1db-446b-8c98-030203076c30.png)
![image](https://user-images.githubusercontent.com/10299707/174696603-6a5f857b-7a5d-4b8b-9985-6bf4089a1c81.png)


I feel the need for something similar to the apply domain widget, but for the configurations of the inputs, the "Treat variables with the same name as the same variable, even if they are computed using different formulae" doesn't really change anything in this case for some reason

The easiest way would be to work with a single dataset merging them beforehand, but i like to keep datasets separate as when the project grows i don't want to compute every workflow for "unrelated" datasets

**Reply by ajdapretnar** [reply]
You just didn't put Fruit in one data table as a target variable, that's all.

**Reply by Kiskadee-dev** [reply]
I'm only suggesting to have a way to apply those configurations to all data tables instead of one by one as it's prone to mistakes... especially when you have many inputs with many variables
